### DISCUSSION TITLE
"cast send" tip: Error, Failed to create wallet from the private key..

### DISCUSSION THREAD

**Discussion by NokivaF** [Cyfrin/foundry-full-course-cu#24] [Show and tell]
First of all, thanks Patrick, I promise I'll use these powers in order to make people like you emerge above the non-sense out there. It's time to make the real (and overall, RELEVANT) talent be noticed a looot more.

Anyway, I encountered this issue in my machine when working around the minute [7:58](https://youtu.be/umepbfKp5rI?list=PL4Rj_WH6yLgWe7TxankiqkrkVKXIwOP42&t=28676) of the first part of the course, and those who use WSL or a UNIX OS might find it too. It has a quick fix, so first install this (or something alike) package to convert the .env file to UNIX format:

```linux
sudo apt install dos2unix
```

Now just pass the .env file through it, and then load it with "source .env" in order to avoid the error, and all of that **before** running the "cast" command. Remember, every time you kill the terminal, you have to repeat the process.

I found the answer [here.](https://github.com/foundry-rs/foundry/issues/2086#issue-1281692103)

Keep going, you can do it!!!

**Comment by cromewar** [comment]
Hello @NokivaF thanks for reaching out with this solution, a question for you where are you storing the project folders? directly on the WSL home or on your Windows Drive?

**Reply by NokivaF** [reply]
> Hello @NokivaF thanks for reaching out with this solution, a question for you where are you storing the project folders? directly on the WSL home or on your Windows Drive?

Hi there. I store it all within the Linux filesystem, I don't mix anything and besides Linux is more flexible to me.

**Comment by alymurtazamemon** [comment]
@NokivaF Please select the `Show and tell` category for this discussion. Thanks

**Reply by cromewar** [reply]
Changed!
